visualization/pg_joins.js

```diff
@@ -0,0 +1,197 @@
+    const res= (await client.query(query));
+    return res.rows[0]["id"]
+  }
+  catch(e){
+    console.log(e);
+  }
+}
+
+const fetch_specieId= async (client, specie)=>{
+  try{
+    const res= (await client.query('SELECT id FROM species WHERE scientificname = $1', [specie]));
+    return res.rows[0]["id"]
+  }
+  catch(e){
+    console.log(e);
+  }
+}
+
+
+const getTable1=async (cultivars, sites_cultivars, sites) => {
+  const client = await pool.connect()
+  try{
+    await client.query('BEGIN')
+    let dat= cultivars.map(async data =>{ 
+        let specie=data.pop();
+        let specie_id = await fetch_specieId(client,specie)
+        data.push(parseInt(specie_id))
+        return data
+      })
+    let t1= ((await client.query("SELECT UTC_NOW()")).rows[0]["utc_now"]);
+    let res=  Promise.all(dat).then(async (data)=>{ 
+                const query1 = format('INSERT INTO cultivars(name,ecotype,notes,specie_id) VALUES %L',data)
+                await client.query(query1)
+              }).catch((e)=> console.log(e))
+              .then(async()=>{
+                  sites.map((site)=>{
+                  site.pop()
+                  site.pop()
+                  site.pop()
+                  site.push("some text")
+                })
+                const query3 = format('INSERT INTO sites(sitename,city,state,country,notes,time_zone,soil,soilnotes) VALUES %L',sites)
+                await client.query(query3)
+                  let sites_data=sites_cultivars.map(async data =>{
+                  let cultivar_name= data.shift();
+                  let specie_id= data.shift();
+                  let sitename= data.shift();
+                  let cultivar_id= await fetch_cultivarsId(client, cultivar_name, specie_id);
+                  let site_id= await fetch_sitesId(client,sitename);
+                return [site_id, cultivar_id]
+                  })
+                  return Promise.all(sites_data)
+                })
+                .then(async (sites_data)=>{
+                      const query2 = format('INSERT INTO sites_cultivars(site_id,cultivar_id) VALUES %L',sites_data)
+                      await client.query(query2)
+                  })
+                  .then(async()=>{
+                    let join=format("SELECT name as cultivar, sitename as plot FROM cultivars JOIN sites_cultivars ON cultivars.id = sites_cultivars.cultivar_id AND cultivars.created_at >= timestamptz %L JOIN sites ON sites_cultivars.site_id = sites.id",t1)
+                    res=await client.query(join);                    
+                    await client.query('ROLLBACK')
+                    client.release();                          
+                    return res.rows 
+                  })
+                  .catch((e)=> console.log(e))
+                  return res
+  }
+  catch(e){
+    console.log(e)
+  }
+}
+
+const getTable2=async (experiments, experiments_sites, sites) => {
```

Comment:
Use a more descriptive name than "getTable2" here

Reply:
Thanks! for pointing out, will change it in a future commit.